src/graph.cpp

```diff
@@ -135,17 +145,67 @@ void Graph::compute_distance_matrix() const {
   }
 }
 
-void Graph::add_node(boost::filesystem::path const &input_file,
+boost::filesystem::path Graph::add_node(boost::filesystem::path const &input_file,
                      Hash const &input_hash) {
-  assert(not has_node(input_file));
-#pragma omp critical(graph_)
-  {
+  if(not has_node(input_file)){
+//#pragma omp critical(graph_)
+  //{
+    if (passed_path_.find(input_file.parent_path()) == passed_path_.end()){ //path not parsed yet
+	passed_path_[input_file.parent_path()] = 1;
+	//std::cout << "YOP\n";
+    }
+
+
+    boost::filesystem::path no_path1 = "/.";
+    boost::filesystem::path no_path2 = ".";
+    std::string file = input_file.string();
+
+    if(input_file.parent_path() == no_path1 || input_file.parent_path() == no_path2){
+	//std::cout<<"/. or .\n";
+
+	for (it_p_=passed_path_.begin(); it_p_!=passed_path_.end();it_p_++){
```

Comment:
I don't get why this ``it_p_`` is a member variable, it seems to be used as a local one only.